src/platform/packages/shared/kbn-unified-field-list/src/components/field_item_button/index.tsx

```diff
@@ -7,4 +7,23 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
-export { FieldItemButton, type FieldItemButtonProps } from './field_item_button';
+import React, { Fragment } from 'react';
+import type { FieldItemButtonProps, GenericFieldItemButtonType } from './field_item_button';
+import { FieldListItem } from '../../types';
+
+const Fallback = () => <Fragment />;
+
+const LazyFieldItemButton = React.lazy(() =>
+  import('./field_item_button').then((module) => ({ default: module.FieldItemButton }))
+) as GenericFieldItemButtonType;
+
+function WrappedFieldItemButton<T extends FieldListItem>(props: FieldItemButtonProps<T>) {
+  return (
+    <React.Suspense fallback={<Fallback />}>
+      <LazyFieldItemButton {...props} />
+    </React.Suspense>
+  );
+}
+
+export type { FieldItemButtonProps };
+export const FieldItemButton = WrappedFieldItemButton;
```

Comment:
Could we export a separate lazy version of this component instead? Lens imports it directly, but Discover uses it through `UnifiedFieldListSidebarContainer`, and I think this would add an unnecessary async bundle to that component.

On a side note, I like using the `withSuspense` util for lazy components, but either way is fine.

Reply:
I've moved the async loading into Lens itself and reverted these changes with https://github.com/elastic/kibana/pull/211921/commits/726a3c06642284fd310998d80bb73455ffc96bd5

I've looked into `withSuspense` and `dynamic` too but both have the same root problem of any lazy loaded component: the generic type is lost so a force cast is required afterworse.

Reply:
Thanks! And TIL, I didn't realize lazy loaded components lose their generic type info like that 👍